.changeset/curvy-planets-serve.md

```diff
@@ -0,0 +1,5 @@
+---
+'xstate': minor
+---
+
+Upgraded TypeScript to 3.8.3. This also addresses #1015 and simplifies the `TransitionConfigArray` and `TransitionConfigMap` types.
```

Comment:
"Upgraded TypeScript to 3.8.3" - this part is not really relevant to our consumers. I would just focus here on simplifying those 2 types and mentioning what it is supposed to fix (excessive type instantiation reported by TS). You could also link to the mentioned issue.

```suggestion
Simplify the `TransitionConfigArray` and `TransitionConfigMap` types In order to fix excessively deep type instantiation TypeScript reports. This addresses [#1015](https://github.com/davidkpiano/xstate/issues/1015).
```

Reply:
The TypeScript version is definitely relevant to many users.